src/Controllers/InventoryController.php

```diff
@@ -1,149 +1,111 @@
-                'warehouseId'
-            ]
-        ],
-        'variationLinkMarketplace' => [
-            'marketplaceId'
-        ],
-        'variationWarehouseList' => [
-            'variationId',
-            'warehouseId',
-        ],
-        'itemDescription' => [
-            'name1'
-        ]
-    ];
-    $filters = [
-        'variationStock.wasUpdatedBetween' => [
-            'timestampFrom' => time() - 4800,
-            'timestampTo' => time(),
-        ],
-        'variationVisibility.isVisibleForMarketplace' => [
-            'mandatoryAllMarketplace' => [$data],
-            'mandatoryOneMarketplace' => []
-        ]
-    ];
-    $result2 = $itemDataLayerRepository->search($resultFields, $filters);
-
-    return json_encode(
-        [
-            'result' => $result2->toArray()
-        ]
-    );
+  public function sync()
+  {
+    $this->logger->debug(TranslationHelper::getLoggerKey(self::LOG_KEY_CONTROLLER_IN), [
+      'additionalInfo' => [
+        'full' => $this->full,
+        'payloadIn' => ''
+      ],
+      'method'         => __METHOD__
+    ]);
+
+    $this->inventoryUpdateService->sync($this->full, true);
+    // set manual flag so that we know where sync request came from
+    return $this->getState();
   }
 
   /**
-   * @param Request $request
-   *
-   * @return false|string
+   * @return string
    */
-  public function getItem(Request $request) {
-    $data = $request->input('data');
-    /**
-     * @var ItemRepositoryContract $itemRepositoryContract
-     */
-    $itemRepositoryContract = pluginApp(ItemRepositoryContract::class);
-    $item = $itemRepositoryContract->show($data);
-
-    return json_encode($item);
+  public function getState()
+  {
+    $dataOut = json_encode($this->inventoryStatusService->getServiceState($this->full));
+
+    $this->logger->debug(TranslationHelper::getLoggerKey(self::LOG_KEY_CONTROLLER_OUT), [
+      'additionalInfo' => [
+        'full' => $this->full,
+        'payloadOut' => $dataOut
+      ],
+      'method'         => __METHOD__
+    ]);
+
+    return $dataOut;
   }
 
   /**
-   * @return false|string
-   * @throws \Exception
+   * @return void
    */
-  public function sync() {
-    /**
-     * @var InventoryUpdateService $inventoryUpdateService
-     */
-    $inventoryUpdateService = pluginApp(InventoryUpdateService::class);
+  public function clearState()
+  {
+    $this->logger->debug(TranslationHelper::getLoggerKey(self::LOG_KEY_CONTROLLER_IN), [
+      'additionalInfo' => [
+        'full' => $this->full,
+        'payloadIn' => ''
+      ],
+      'method'         => __METHOD__
+    ]);
+
+    $this->inventoryStatusService->clearState($this->full, true);
 
-    return json_encode($inventoryUpdateService->sync());
+    $this->logger->debug(TranslationHelper::getLoggerKey(self::LOG_KEY_CONTROLLER_OUT), [
+      'additionalInfo' => [
+        'full' => $this->full,
+        'payloadOut' => ''
+      ],
+      'method'         => __METHOD__
+    ]);
```

Comment:
What is the need for both `logger->debug` for `LOG_KEY_CONTROLLER_IN` and `LOG_KEY_CONTROLLER_OUT`?

Reply:
`IN`: call came into the back end
`OUT`: done processing, returning back to client